pkg/util/json/json.go

```diff
@@ -1124,6 +1129,91 @@ func (j jsonObject) Exists(s string) (bool, error) {
 	return v != nil, nil
 }
 
+func (j jsonNull) StripNulls() (JSON, bool, error) {
+	return j, false, nil
+}
+func (j jsonTrue) StripNulls() (JSON, bool, error) {
+	return j, false, nil
+}
+func (j jsonFalse) StripNulls() (JSON, bool, error) {
+	return j, false, nil
+}
+func (j jsonNumber) StripNulls() (JSON, bool, error) {
+	return j, false, nil
+}
+func (j jsonString) StripNulls() (JSON, bool, error) {
+	return j, false, nil
+}
+func (j jsonArray) StripNulls() (JSON, bool, error) {
+	for i, e := range j {
+		json, needToStrip, err := e.StripNulls()
+		if err != nil {
+			return nil, false, err
+		}
+		if needToStrip {
+			// Cannot return the original content, need to return the result
+			// with new JSON array.
+			newArr := make(jsonArray, 0, len(j))
+			newArr = append(append(newArr, j[:i]...), json)
+			for _, elem := range j[i+1:] {
+				if json, _, err = elem.StripNulls(); err != nil {
+					return nil, false, err
+				}
+				newArr = append(newArr, json)
+			}
+			return newArr, true, nil
+		}
+	}
+	return j, false, nil
+}
+func (j jsonObject) StripNulls() (JSON, bool, error) {
+	for i, e := range j {
+		var json JSON
+		var err error
+		needToStrip := false
+		hasNullValue := e.v.Type() == NullJSONType
+		if !hasNullValue {
+			json, needToStrip, err = e.v.StripNulls()
+			if err != nil {
+				return nil, false, err
+			}
+		}
+		if hasNullValue || needToStrip {
+			// Cannot return the original content, need to return the result
+			// with new JSON object.
+			numNotNulls := i
+			for _, elem := range j[i:] {
+				if elem.v.Type() != NullJSONType {
+					numNotNulls++
+				}
+			}
+			// Use number of fields not having null value to construct newObj
+			// so that no need to grow the capacity of newObj.
+			newObj := make(jsonObject, 0, numNotNulls)
```

Comment:
I'm not sure the additional complexity here in doing these `numNotNulls` checks is worth the benefit, unless you have a benchmark to show otherwise?